api/core/workflow/nodes/agent/agent_node.py

```diff
@@ -251,7 +251,15 @@ def _generate_agent_parameters(
                                 prompt_message.model_dump(mode="json") for prompt_message in prompt_messages
                             ]
                     value["history_prompt_messages"] = history_prompt_messages
-                    value["entity"] = model_schema.model_dump(mode="json") if model_schema else None
+                    if model_schema:
+                        # remove structured output feature to support old version agent plugin
+                        if model_schema.features:
```

Comment:
I recommend extracting this logic into a separate function (or perhaps a method on `ProviderModel`) named `remove_unsupported_model_features_for_old_version`.

```python
if model_schema.features:
    for feature in model_schema.features:
        if feature.value not in AgentOldVersionModelFeatures:
            model_schema.features.remove(feature)
```

Doing so will help simplify the compatibility logic within `AgentNode` and improve code readability and reusability.